src/main/java/com/zaxxer/hikari/pool/PoolEntry.java

```diff
@@ -90,7 +90,9 @@ void setFutureEol(final ScheduledFuture<?> endOfLife)
 
    Connection createProxyConnection(final ProxyLeakTask leakTask, final long now)
    {
-      return ProxyFactory.getProxyConnection(this, connection, openStatements, leakTask, now, hikariPool.isAutoCommit());
+      ProxyConnection pc = ProxyFactory.getProxyConnection(this, connection, openStatements, leakTask, now);
+      hikariPool.initProxyConnection(pc);
```

Comment:
I suspect that this is going to cause a performance hit, because this is in the hot path and the ``init()`` method in ``ProxyConnection`` introduces and additional dereference and method invocation.

I initially did it like this, but eventually introduced it as an additional stack parameter to the constructor to reduce the overhead.